conformance/lang/expresssions/literals/byte_array_literal.balt

```diff
@@ -0,0 +1,480 @@
+Label: byte-array-literal, function-call-expr
+
+public function main() {
+    testByteArray(base16 `aaabcfccadafcd341a4bdfabcd8912df`); // @output [170,171,207,204,173,175,205,52,26,75,223,171,205,137,18,223]
+}
+
+function testByteArray(byte[] byteArray) {
+    io:println(byteArray);
+}
+
+Test-Case: output
+Description: Test base16 byte array literal in type test expression.
+Label: byte-array-literal, is-expr
+
+public function main() {
+    io:println(base16 `aaabcfccadafcd341a4bdfabcd8912df` is byte[]); // @output true
+}
+
+Test-Case: output
+Description: Test base16 byte array literal with method call.
+Label: byte-array-literal, method-call-expr, value:toString, value:toBalString
+
+public function main() {
+    io:println(base16 `aaabcfccadafcd341a4bdfabcd8912df`.toString()); // @output [170,171,207,204,173,175,205,52,26,75,223,171,205,137,18,223]
+    io:println(base16 `aaabcfccadafcd341a4bdfabcd8912df`.toBalString()); // @output [170,171,207,204,173,175,205,52,26,75,223,171,205,137,18,223]
+}
+
+Test-Case: output
+Description: Test base16 byte array literal in if else statement.
+Label: byte-array-literal, if-else-stmt, array:length
+
+public function main() {
+    if base16 `af12daa0`.length() > 0 {
+        io:println(true); // @output true
+    }
+}
+
+Test-Case: output
+Description: Test base16 byte array literal in match statement.
+Label: byte-array-literal, match-stmt, var
+
+public function main() {
+    match base16 `abcd` {
+        var v => {
+            io:println(v); // @output [171,205]
+        }
+    }
+}
+
+Test-Case: output
+Description: Test base16 byte array literal with member access.
+Label: byte-array-literal, member-access-expr
+
+public function main() {
+    io:println(base16 `00FFFF`[0]); // @output 0
+    io:println(base16 `00FFFF`[1]); // @output 255
+    io:println(base16 `00FFFF`[2]); // @output 255
+}
+
+Test-Case: output
+Description: Test base16 byte array literal in structural constructors.
+Label: byte-array-literal, list-constructor-expr, mapping-constructor-expr, table-constructor-expr
+
+public function main() {
+    io:println([base16 `aabb`, base16 `ccdd`]); // @output [[170,187],[204,221]]
+    io:println({byteArray1: base16 `aabb`, byteArray2: base16 `abcd`}); // @output {"byteArray1":[170,187],"byteArray2":[171,205]}
+    io:println(table [{ id:1, byteArray: base16 `abcd`}]); // @output [{"id":1,"byteArray":[171,205]}]
+}
+
+Test-Case: output
+Description: Test base16 byte array literal in return statement.
+Label: byte-array-literal, return-stmt
+
+public function main() {
+    io:println(getByteArray()); // @output [171,205]
+}
+
+function getByteArray() returns byte[] {
+    return base16 `abcd`;
+}
+
+Test-Case: output
+Description: Test base16 byte array literal with foreach statement.
+Label: byte-array-literal, foreach-stmt
+
+public function main() {
+    foreach byte i in base16 `abcd` {
+        io:println(i); // @output [171,205]
+    }
+}
+
+Test-Case: error
```

Comment:
This test seems to be failing. Could be due to the extra comment. Can you please verify and raise it on the test runner PR?